upgrade/upgrade-scripts/execute

```diff
@@ -28,7 +28,29 @@ function usage() {
 	exit 2
 }
 
+function wait_for_apt_lock() {
+	# The 'apt' package manager commands will fail if other package manager
+	# already holds the lock. This can fail upgrade workflow, so we try to
+	# mitigate by waiting up to a set timeout and fail only if timeout is reached.
+	local timeout=300 # Set timeout to 5 min
+
+	until ! output="$(fuser /var/lib/dpkg/{lock,lock-frontend} 2>&1)" || test $((timeout--)) -eq 0; do
```

Comment:
Would it be clearer if we did something like this?
```
now=$(date +%s)
timeout=$((now + 500))

while [[ $(date +%s) -lt $timeout ]]; do
    output=$(...)
    if [[ -n "$output" ]]; then
        # i think if you already saved in variable you can just do "for x in" instead of streaming it again
        for line in "$output"; do 
             ...
        done
    else
      return
    fi

# if we reached here then we hit timeout
die
```